src/DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp

```diff
@@ -0,0 +1,357 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines function taking the inverse of a rank-2 tensor
+
+#pragma once
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/MakeArithmeticValue.hpp"
+
+/*
```

Comment:
I went with a struct instead of a pair because — until we require C++17 — users would be writing the annoying:
```
Scalar<T> det;
Tensor<T, Symm, complex_index_manipulations<Indices>> inv;
std::tie(det, inv) = determinant_and_inverse(input);
```

But if this doesn't bother you, I can change struct to pair.